Sidebands/SidebandDraw.py

```diff
@@ -0,0 +1,100 @@
+
+          if verb: print("Making plots with preselection",preselection,"and selection",selection)
+               
+          if sideband_title != None and selection+"_"+DATASET not in selections.selection_categories.keys():
+              sel = selections.selection_categories[selection]
+              sel["title"] = sel["title"] + ", " + sideband_title 
+              selections.selection_categories[selection+"_"+DATASET] = sel
+
+            
+          os.system("mkdir -p Plots/png/run_"+runcombo_str+"/"+DATASET+"/"+preselection+"_"+selection)
+          os.system("mkdir -p Plots/pdf/run_"+runcombo_str+"/"+DATASET+"/"+preselection+"_"+selection)
+
+          for variables in VARIABLE_v:
+
+              for binning_def in variables:
+
+                  if verb: print("Making plot of",binning_def[0])
+    
+                  # Check the variable exists in the dataframes, skip this var if it doesn't
+                  found_variable=True
+                  for key in rundata.keys():                  
+                      if binning_def[0] not in rundata[key].columns: found_variable=False
+        
+                  # some binning definitions have more than 4 elements,
+                  # we ignore the last ones for now
+                  binning = hist.Binning.from_config(*binning_def[:4])
+                  
+                  signal_generator = hist.RunHistGenerator(
+                      rundata,
+                      binning,
+                      data_pot=data_pot,
+                      selection=selection,
+                      preselection=preselection,
+                      sideband_generator=None,
+                      uncertainty_defaults=None,
+                  )
+
+                  plotter = rp.RunHistPlotter(signal_generator)
+                  axes = plotter.plot(
+                      category_column="paper_category",
+                      include_multisim_errors=True,
+                      add_ext_error_floor=True,
+                      show_data_mc_ratio=True,
+                      show_chi_square=True,
+                  )
+                  
+                  # Form a unique name for each plot
+                  #pltname = "Plots/pdf/run_" + runcombo_str + "/" + DATASET + "_" + binning_def[0] + "_run" + runcombo_str + "_" + selection + ".pdf"
+                  pltname = "Plots/pdf/run_" + runcombo_str + "/" + DATASET + "/" + preselection + "_" + selection + "/" +\
+                            DATASET + "_" + binning_def[0] + "_run" + runcombo_str + "_" + preselection + "_" + selection + ".pdf"
+                  plt.savefig(pltname)
+                  pltname = "Plots/png/run_" + runcombo_str + "/" + DATASET + "/" + preselection + "_" + selection + "/" +\
+                            DATASET + "_" + binning_def[0] + "_run" + runcombo_str + "_" + preselection + "_" + selection + ".png"
+                  plt.savefig(pltname)
+                  plt.close()     
+              
+      del rundata
```

Comment:
Should be no need to delete variables manually if you are going to overwrite them later. The garbage collection will notice that no objects hold any references to the original data anymore and take care of it.